microbench/src/main/java/io/netty/microbench/internal/PrivilegedSocketOperationsBenchmark.java

```diff
@@ -0,0 +1,193 @@
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.channels.ServerSocketChannel;
+import java.security.AccessController;
+import java.security.NoSuchAlgorithmException;
+import java.security.Policy;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.security.URIParameter;
+import java.util.concurrent.TimeUnit;
+
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.SECONDS)
+public class PrivilegedSocketOperationsBenchmark extends AbstractMicrobenchmark {
+
+    @State(Scope.Benchmark)
+    public static class SecurityManagerInstalled {
+
+        @Setup
+        public void setup() throws IOException, NoSuchAlgorithmException, URISyntaxException {
+            final URI policyFile = PrivilegedSocketOperationsBenchmark.class.getResource("/jmh-security.policy")
+                    .toURI();
+            Policy.setPolicy(Policy.getInstance("JavaPolicy", new URIParameter(policyFile)));
+            System.setSecurityManager(new SecurityManager());
+        }
+
+        @TearDown
+        public void tearDown() throws IOException {
+            System.setSecurityManager(null);
+        }
+    }
+
+    @State(Scope.Benchmark)
+    public static class SecurityManagerEmpty {
+
+        @Setup
+        public void setup() throws IOException, NoSuchAlgorithmException, URISyntaxException {
+            System.setSecurityManager(null);
+        }
+    }
+
+    @Benchmark
+    public ServerSocketChannel testWithSMNoPrivileged(final SecurityManagerInstalled sm) throws IOException {
+        final ServerSocketChannel ssc = ServerSocketChannel.open();
+        ssc.socket().bind(null);
+        ssc.configureBlocking(false);
+        ssc.accept();
+        ssc.close();
+        return ssc;
+    }
+
+    @Benchmark
+    public ServerSocketChannel testWithSM(final SecurityManagerInstalled sm) throws IOException {
+        try {
+            final ServerSocketChannel ssc = AccessController.doPrivileged(
+                    new PrivilegedExceptionAction<ServerSocketChannel>() {
+                        @Override
+                        public ServerSocketChannel run() throws Exception {
+                            final ServerSocketChannel ssc = ServerSocketChannel.open();
+                            ssc.socket().bind(null);
+                            ssc.configureBlocking(false);
+                            ssc.accept();
+                            return ssc;
+                        }
+                    });
+            ssc.close();
+            return ssc;
+        } catch (final PrivilegedActionException e) {
+            throw (IOException) e.getCause();
+        }
+    }
+
+    @Benchmark
+    public ServerSocketChannel testWithSMWithNullCheck(final SecurityManagerInstalled sm) throws IOException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "sm". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1172)